### DISCUSSION TITLE
local extinction or extirpation

### DISCUSSION THREAD

**Discussion by alanrogers** [tskit-dev/msprime#2292] [Q&A] [answered]
I'd like to simulate a model in which demes occasionally go extinct and are immediately replaced with immigrants drawn from another deme. In one version of this model (Slatkin's 1977 "propagule model") all these immigrants come from a single other deme. In reverse time, such events look like mass migration, with all residents of a deme suddenly moving to a different deme.

I can't find any indication that msprime implements such a model. Have I missed something?

Thanks, Alan Rogers

**Comment by jeromekelleher** [comment, marked as answer]
There's no support for demes going extinct randomly as part of the simulation, but you could randomly generate a demographic model first under the process in question, and then run the simulation under this demography. If you do this independently across replicates, I think that should do what you want? I'm happy to give some code if you'd like some pointers in setting it up.

**Reply by alanrogers** [reply]
Thanks. That's what I suspected. If I generate a random demographic model as you suggest, it will have to be finite, and I'll need to check whether any of the gene trees in my simulation are deeper than that finite limit. What is the best way to do that?

**Reply by jeromekelleher** [reply]
Something like ``np.any(ts.nodes_time > demographic_model_time_limit)`` would do the trick (where ts is the result of your simulations)

**Reply by alanrogers** [reply]
Many thanks.
